PandorasBox/Features/UI/WorkshopHelper.cs

```diff
@@ -0,0 +1,591 @@
+                TaskManager.Enqueue(() => SelectedCycle -= 1);
+            }
+            ImGui.SameLine();
+            if (ImGui.Button("Next"))
+            {
+                TaskManager.Enqueue(() => OpenCycle(SelectedCycle + 1));
+                TaskManager.Enqueue(() => SelectedCycle += 1);
+            }
+
+            ImGui.Text("Select Workshops");
+            for (var i = 0; i < Workshops.Count; i++)
+            {
+                var configValue = Workshops[i];
+                if (ImGui.Checkbox($"{i + 1}", ref configValue)) { Workshops[i] = configValue; }
+                if (i != Workshops.Count - 1) ImGui.SameLine();
+            }
+            ImGui.SameLine();
+            if (ImGui.Button("Deselect"))
+                Workshops.Keys.ToList().ForEach(key => Workshops[key] = false);
+
+            try
+            {
+                if (ImGui.Button("Execute Schedule"))
+                {
+                    CurrentWorkshop = Workshops.FirstOrDefault(pair => pair.Value).Key;
+                    var restDays = GetCurrentRestDays();
+                    if (restDays.Contains(SelectedCycle - 1))
+                        PrintPluginMessage("Selected cycle is a rest day. Cannot schedule on a rest day.");
+                    else
+                        ScheduleList();
+                }
+            }
+            catch (Exception e) { PluginLog.Log(e.ToString()); return; }
+        }
+
+        public static unsafe Vector2 GetNodePosition(AtkResNode* node)
+        {
+            var pos = new Vector2(node->X, node->Y);
+            var par = node->ParentNode;
+            while (par != null)
+            {
+                pos *= new Vector2(par->ScaleX, par->ScaleY);
+                pos += new Vector2(par->X, par->Y);
+                par = par->ParentNode;
+            }
+
+            return pos;
+        }
+
+        public static unsafe Vector2 GetNodeScale(AtkResNode* node)
+        {
+            if (node == null) return new Vector2(1, 1);
+            var scale = new Vector2(node->ScaleX, node->ScaleY);
+            while (node->ParentNode != null)
+            {
+                node = node->ParentNode;
+                scale *= new Vector2(node->ScaleX, node->ScaleY);
+            }
+
+            return scale;
+        }
+
+        internal static List<Item> ScheduleImport(List<string> rawItemStrings)
+        {
+            List<Item> items = ParseItems(rawItemStrings);
+            CopiedSchedule = items;
+            return CopiedSchedule;
+        }
+
+        public static List<Item> ParseItems(List<string> itemStrings)
+        {
+            List<Item> items = new List<Item>();
+            foreach (var itemString in itemStrings)
+            {
+                bool matchFound = false;
+                foreach (var craftable in Craftables)
+                {
+                    string craftableNoPrefix = craftable.Name.Replace("Isleworks ", "");
```

Comment:
There are 2 entries that don't start with "Isleworks" (both 6h): Isleberry Jam, and Islefish Pie. Those strings should probably be removed as well. There's also Bronze Sheep (8h) which has no prefix, but that shouldn't cause a problem.

Reply:
fixed in #82c86f3

Reply:
Don't forget this will likely only work on the English client. Probably best to look at alternatives that don't involve replacing hard-coded words where possible.

Reply:
Keep in mind this is specifically for the Overseas Casuals and I've made no allusions it's anything else. They don't do other languages afaik and a generic importer would require an entirely different matching approach because this matches the craftables against the clipboard and not the other way around (that would require a really good junk filtering method).